.buildkite/rootfs_images/rootfs_utils.jl

```diff
@@ -74,12 +88,59 @@ function debootstrap(name::String; release::String="buster", variant::String="mi
         end
         run(`sudo chroot --userspec=$(getuid()):$(getgid()) $(rootfs) locale-gen`)
     end
+end
 
-    # Archive it into a `.tar.gz` file
-    @info("Archiving", tarball_path, artifact_hash)
-    archive_artifact(artifact_hash, tarball_path)
+# Helper structure for installing alpine packages that may or may not be part of an older Alpine release
+struct AlpinePackage
+    name::String
+    repo::Union{Nothing,String}
 
-    return tarball_path
+    AlpinePackage(name, repo=nothing) = new(name, repo)
+end
+function repository_arg(repo)
+    if repo === nothing
+        return String[]
+    end
+    if startswith(repo, "https://")
+        return ["--repository=$(repo)"]
+    end
+    return ["--repository=http://dl-cdn.alpinelinux.org/alpine/$(repo)/main"]
+end
+repository_arg(pkg::AlpinePackage) = repository_arg(pkg.repo)
+
+
+function alpine_bootstrap(name::String; release::VersionNumber=v"3.13.5", variant="minirootfs",
+                          packages::Vector{AlpinePackage}=AlpinePackage[], force::Bool=false)
+    return create_rootfs(name; force) do rootfs
+        rootfs_url = "https://github.com/alpinelinux/docker-alpine/raw/v$(release.major).$(release.minor)/x86_64/alpine-$(variant)-$(release)-x86_64.tar.gz"
+        @info("Downloading Alpine rootfs", url=rootfs_url)
+        rm(rootfs)
+        Pkg.Artifacts.download_verify_unpack(rootfs_url, nothing, rootfs; verbose=true)
+
+        # Remove special `dev` files, take ownership and write out some standard files
+        cleanup_dev_perms_resolv(rootfs)
+
+        # Write out rootfs-info to contain a minimally-identifying string
+        open(joinpath(rootfs, "etc", "rootfs-info"), write=true) do io
+            write(io, """
+            rootfs_type=alpine
+            release=$(release)
+            variant=$(variant)
+            packages=$(join([pkg.name for pkg in packages], ","))
+            build_date=$(Dates.now())
+            """)
+        end
+
+        # Generate one `apk` invocation per repository
+        repos = unique([pkg.repo for pkg in packages])
+        for repo in repos
+            apk_cmd = `apk add --no-chown $(repository_arg(repo))`
+            for package in filter(pkg -> pkg.repo == repo, packages)
+                apk_cmd = `$(apk_cmd) $(package.name)`
+            end
+            run(`sudo chroot --userspec=$(getuid()):$(getgid()) $(rootfs) $(apk_cmd)`)
+        end        
```

Comment:
```suggestion
        end
```
This line is what's making the whitespace check sad